src/frontend/next/src/components/DeleteFeedDialogButton.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState, useRef } from 'react';
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogActions,
+  DialogContentText,
+  DialogTitle,
+  IconButton,
+} from '@material-ui/core';
+import { Delete } from '@material-ui/icons';
+import { makeStyles } from '@material-ui/core/styles';
+import { Feed } from '../interfaces';
+import useSiteMetadata from '../hooks/use-site-metadata';
+
+type DeleteFeedDialogButtonProps = {
+  feed: Feed;
+  deletionCallback: (id: number) => void;
+};
+
+const useStyles = makeStyles(() => ({
+  button: {
+    padding: '3px 0 3px 0',
+  },
+}));
+
+const DeleteFeedDialogButton = ({ feed, deletionCallback }: DeleteFeedDialogButtonProps) => {
+  const { id, url } = feed;
+  const classes = useStyles();
+  const { telescopeUrl } = useSiteMetadata();
+  const [open, setOpen] = useState<boolean>(false);
+
+  const deleteBtnRef = useRef<HTMLButtonElement>(null);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const disableBtn = () => {
+    if (deleteBtnRef.current !== null) {
+      deleteBtnRef.current.setAttribute('disabled', 'disabled');
+    }
+  };
+
+  const removeFeed = async () => {
+    console.log(`Removing feed hosted at URL ${url}...`);
+    try {
+      const response = await fetch(`${telescopeUrl}/feeds/${id}`, { method: 'DELETE ' });
```

Comment:
Stupid question but `'DELETE '` has an extra space in it, is it needed?

Reply:
Definitely not, that's my fast typing lol